src/libstore/globals.cc

```diff
@@ -68,7 +61,12 @@ Settings::Settings()
     sandboxPaths = tokenizeString<StringSet>("/bin/sh=" SANDBOX_SHELL);
 #endif
 
-    allowedImpureHostPrefixes = tokenizeString<StringSet>(DEFAULT_ALLOWED_IMPURE_PREFIXES);
+
+/* chroot-like behavior from Apple's sandbox */
+#if __APPLE__
+    sandboxPaths = tokenizeString<StringSet>("/System/Library/Frameworks /System/Library/PrivateFrameworks /bin/sh /bin/bash /private/tmp /private/var/tmp /usr/lib");
```

Comment:
I would _really_ prefer not to need bash here (https://github.com/NixOS/nix/issues/3223) but other than either disallowing `/bin/sh` completely or having nix depend on something like fuse bindfs I don't think we can work around it. 😕

Reply:
I really wish there was some way to say "allow /bin/bash but only when executed by /bin/sh" 🤷‍♀️